2018/proposals/gensim Aneesh Joshi Similarity Learning.md

```diff
@@ -0,0 +1,399 @@
+
+
+## Introduction
+Deep Learning essentially operates on allowing the model to find a suitable vector representation for a given input signal. Thus, it lends itself well to the task of Similarity Learning where we want to find the similarity between two documents without explicity deciding the way it is represented. Similarity Learning has several use cases like:
+
+- document retrieval
+- question answering
+- conversational response ranking
+- paraphrase identification 
+
+<table>
+  <tr>
+    <th width=30%, bgcolor=#999999 >Tasks</th> 
+    <th width=20%, bgcolor=#999999>Text 1</th>
+    <th width="20%", bgcolor=#999999>Text 2</th>
+    <th width="20%", bgcolor=#999999>Objective</th>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Paraphrase Indentification </td>
+    <td align="center", bgcolor=#eeeeee> string 1 </td>
+    <td align="center", bgcolor=#eeeeee> string 2 </td>
+    <td align="center", bgcolor=#eeeeee> classification </td>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Textual Entailment </td>
+    <td align="center", bgcolor=#eeeeee> text </td>
+    <td align="center", bgcolor=#eeeeee> hypothesis </td>
+    <td align="center", bgcolor=#eeeeee> classification </td>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Question Answer </td>
+    <td align="center", bgcolor=#eeeeee> question </td>
+    <td align="center", bgcolor=#eeeeee> answer </td>
+    <td align="center", bgcolor=#eeeeee> classification/ranking </td>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Conversation </td>
+    <td align="center", bgcolor=#eeeeee> dialog </td>
+    <td align="center", bgcolor=#eeeeee> response </td>
+    <td align="center", bgcolor=#eeeeee> classification/ranking </td>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Information Retrieval </td>
+    <td align="center", bgcolor=#eeeeee> query </td>
+    <td align="center", bgcolor=#eeeeee> document </td>
+    <td align="center", bgcolor=#eeeeee> ranking </td>
+  </tr>
+</table>
+-- [1]
+
+## Technical Details
+Most of the Similarity Learning techniques make use of some form of neural network which is trained in order to reduce the error for a given task. Depending on the way the inputs, outputs and loss functions are configured, we get broadly 4 ways of using Similarity Learning:
+
+**1. Regresssion Similarity Learning:** given `document1` and `document2` we need a `similiarity measure`
+
+**2. Classification Similarity Learning:** given `document1` and `document2` we need to classify them as similar or disimilar
+
+**3. Ranking Similarity Learning:** given `document`, `similar-document` and `disimilar-document`, we want to learn a `similarity-function` such that `similarity-function(document, similar-document)  > similarity-function(document, disimilar-document)`
+
+**4. Locality Sensitivity Hashing:** hashes input items so that similar items map to the same "buckets" in memory with high probability.

```

Comment:
Looks strange in this context (LSH is an unsupervised method). Also, it will be pretty nice if you add some unsupervised/semi-supervised stuff (if we have no labels or have, but too small number)

Reply:
still no semi-supervised

Reply:
Oops!
I forgot about that.
Will make additions. I hope it won't be too late.